### PR TITLE
fix controller: fix the bug which the controller cannot start due to the format error of the third-party CRD 

### PR DESCRIPTION
**Pull request by yangsoon**
Refer to https://github.com/oam-dev/kubevela/issues/1313 , the format error of third-party CRD will lead to vela-controller exited. In this case, we need to keep the vela-controller run normally.

### CODE REVIEW

File: pkg/dsl/definition/package.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func NewPackageDiscover(config *rest.Config) (*PackageDiscover, error) {
 		pkgKinds: make(map[string][]VersionKind),
 	}
 	if err = pd.RefreshKubePackagesFromCluster(); err != nil {
-		return nil, err
+		return pd, err
```
Comment (wonderflow) [inline pkg/dsl/definition/package.go:73]: will the pd be nil?

Comment (yangsoon) [reply pkg/dsl/definition/package.go:73]: `pd` will be nil, i will add some check and prevent it

**Comment by yangsoon** [comment]
I add an new error type `CUEParseErr` to represent the error when cue parse k8s openapiv2, When we start vela-controller,
If the packageDiscover fails to initialize due to an error in parsing the json schema, it will not affect the startup of the controller.

**Review by wonderflow — APPROVED** [review]
LGTM

File: pkg/dsl/definition/package.go [hunk 2]
```diff
@@ -59,6 +62,25 @@ type VersionKind struct {
...
+	return fmt.Sprintf("%s: %s", cueErr.errType, cueErr.err.Error())
+}
+
+// IsCUEParseErr returns true if the specified error is CUEParseError type.
+func IsCUEParseErr(err error) bool {
+	return errors.As(err, &CUEParseError{})
+}
+
 // NewPackageDiscover will create a PackageDiscover client with the K8s config file.
 func NewPackageDiscover(config *rest.Config) (*PackageDiscover, error) {
```
Comment (ryanzhang-oss) [inline pkg/dsl/definition/package.go:85]: better to have a UT to make sure it works

Comment (wonderflow) [reply pkg/dsl/definition/package.go:85]: yes, ping @yangsoon

Comment (yangsoon) [reply pkg/dsl/definition/package.go:85]: got it i will add UT

**Pull request merged**